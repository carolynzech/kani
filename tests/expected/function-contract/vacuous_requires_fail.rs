// Copyright Kani Contributors
// SPDX-License-Identifier: Apache-2.0 OR MIT
// kani-flags: -Zfunction-contracts

// See https://github.com/model-checking/kani/issues/2793
#[kani::requires(a > 5)]
#[kani::requires(a < 4)]
#[kani::ensures(|result| *result == a)]
fn buggy(a: u32) -> u32 {
    panic!("This code is never tested")
}

#[kani::proof_for_contract(buggy)]
fn prove_buggy() {
    let x: u32 = kani::any();
    buggy(x);
}
