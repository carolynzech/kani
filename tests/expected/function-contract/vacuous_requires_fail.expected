Failed Checks: The contract's precondition (i.e., the conjunction of the #[requires(...)] clauses' bodies) is satisfiable.

VERIFICATION:- FAILED

Verification failed for - prove_buggy